ecommerce/views/v0/__init__.py

```diff
@@ -0,0 +1,79 @@
+"""
+MITxOnline ecommerce views
+"""
+
+import pytz
+import datetime
+
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.viewsets import ReadOnlyModelViewSet
+from rest_framework.pagination import LimitOffsetPagination
+from django.db.models import Q, Count
+
+from courses.models import (
+    CourseRun,
+    Course,
+    Program,
+)
+
+from ecommerce.serializers import ProductSerializer
+from ecommerce.models import Product
+
+class ProductsPagination(LimitOffsetPagination):
+    default_limit = 2
+    limit_query_param = 'l'
+    offset_query_param = 'o'
+    max_limit = 50
+
+class ProductViewSet(ReadOnlyModelViewSet):
+    serializer_class = ProductSerializer
+    pagination_class = ProductsPagination
+
+    def get_queryset(self):
+        now = datetime.datetime.now(tz=pytz.UTC)
+
+        expired_courseruns = CourseRun.objects.filter(
+            enrollment_end__lt=now
+        ).values_list("id", flat=True)
+
+        expired_courses = (
+            Course.objects.annotate(
+                runs=Count("courseruns", filter=~Q(courseruns__in=expired_courseruns))
+            )
+            .filter(runs=0)
+            .values_list("id", flat=True)
+        )
+
+        expired_programs = (
+            Program.objects.annotate(
+                valid_runs=Count(
+                    "programruns",
+                    filter=Q(programruns__end_date__gt=now)
+                    | Q(programruns__end_date=None),
+                )
+            )
+            .filter(
+                Q(programruns__isnull=True)
+                | Q(valid_runs=0)
+                | Q(courses__in=expired_courses)
+            )
+            .values_list("id", flat=True)
+            .distinct()
+        )
+
+        return (
+            Product.objects.exclude(
```

Comment:
Are we sure we want to exclude past runs of a product? If this queryset is returned for users enrollment info I feel like we would want all of them, right? I might be missing on some context though.

Reply:
Just to add context - my interpretation from the issue was that these API calls would be used to display currently-purchasable items, Pulling enrollment data or order history or things along those lines would be handled elsewhere.

Reply:
I think this is ok as-is for now. If we want to add additional filtering based on a user's enrollments I think we'd want to do that with some optional filtering, but I don't think that's worth adding until we have a use case for it.